File: Examples/CSharp/Loading-and-Saving/SaveRangeOfPagesAsPDF.cs
using System.IO;
using Aspose.Note;
using Aspose.Note.Saving;
using System;

namespace Aspose.Note.Examples.CSharp.Loading_Saving
{
    public class SaveRangeOfPagesAsPDF
    {
        public static void Run()
        {
            // ExStart:SaveRangeOfPagesAsPDF
            // ExFor:Document
            // ExFor:Document.Save(System.String, Aspose.Note.Saving.SaveOptions)
            // ExFor:PdfSaveOptions
            // ExFor:SaveOptions.PageIndex
            // ExFor:SaveOptions.PageCount
            // ExSummary:Shows how to save a document in pdf format.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadingAndSaving();

            // Load the document into Aspose.Note.
            Document oneFile = new Document(dataDir + "Aspose.one");

            // Initialize PdfSaveOptions object
            PdfSaveOptions opts = new PdfSaveOptions
                                      {
                                          // Set page index of first page to be saved
                                          PageIndex = 0,

                                          // Set page count
                                          PageCount = 1,
                                      };

            // Save the document as PDF
            dataDir = dataDir + "SaveRangeOfPagesAsPDF_out.pdf";
            oneFile.Save(dataDir, opts);

            // ExEnd:SaveRangeOfPagesAsPDF

            Console.WriteLine("\nOneNote specified page range converted successfully to PDF.\nFile saved at " + dataDir);
        }
    }
}

File: Examples/VisualBasic/Loading-and-Saving/SaveRangeOfPagesAsPDF.vb
Imports System.IO
Imports Aspose.Note
Imports Aspose.Note.Saving

Namespace Loading_Saving
    Public Class SaveRangeOfPagesAsPDF
        Public Shared Sub Run()
            ' ExStart:SaveRangeOfPagesAsPDF
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadingAndSaving()

            ' ExStart:LoadOneNote
            ' Load the document into Aspose.Note.
            Dim oneFile As New Document(dataDir & Convert.ToString("Aspose.one"))
            ' ExEnd:LoadOneNote

            ' Initialize PdfSaveOptions object
            Dim opts As New PdfSaveOptions()

            ' Set page index
            opts.PageIndex = 0

            ' Set page count
            opts.PageCount = 1

            dataDir = dataDir & Convert.ToString("SaveRangeOfPagesAsPDF_out.pdf")

            ' Save the document as PDF
            oneFile.Save(dataDir, opts)
            ' ExEnd:SaveRangeOfPagesAsPDF
            Console.WriteLine(Convert.ToString(vbLf & "OneNote specified page range converted successfully to PDF." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace